tests/unittests/aggregation_tests.py

```diff
@@ -861,6 +861,81 @@ def test_needs_frames(self):
         values = dataset.values(expr1 & expr2)
         self.assertListEqual(values, [False, False])
 
+    @drop_datasets
+    def test_serialize(self):
+        samples = [
+            fo.Sample(
+                filepath="image1.jpg",
+                predictions=fo.Detections(
+                    detections=[
+                        fo.Detection(
+                            label="cat",
+                            bounding_box=[0.1, 0.1, 0.4, 0.4],
+                            confidence=0.7,
+                        ),
+                        fo.Detection(
+                            label="dog",
+                            bounding_box=[0.3, 0.3, 0.5, 0.5],
+                            confidence=0.9,
+                        ),
+                    ]
+                ),
+            ),
+            fo.Sample(
+                filepath="image2.jpg",
+                predictions=fo.Detections(
+                    detections=[
+                        fo.Detection(
+                            label="rabbit",
+                            bounding_box=[0.4, 0.4, 0.2, 0.2],
+                            confidence=0.8,
+                        ),
+                    ]
+                ),
+            ),
+            fo.Sample(
+                filepath="image3.jpg",
+                predictions=fo.Detections(
+                    detections=[
+                        fo.Detection(
+                            label="squirrel",
+                            bounding_box=[0.5, 0.5, 0.5, 0.5],
+                            confidence=0.9,
+                        ),
+                    ]
+                ),
+            ),
+        ]
+
+        dataset = fo.Dataset()
+        dataset.add_samples(samples)
+
+        bbox_area = F("bounding_box")[2] * F("bounding_box")[3]
+
+        aggregations = [
+            fo.Bounds("predictions.detections.confidence"),
+            fo.Count(),
+            fo.Count("predictions.detections"),
+            fo.CountValues("predictions.detections.label"),
+            fo.Distinct("predictions.detections.label"),
+            fo.HistogramValues(
+                "predictions.detections.confidence",
+                bins=50,
+                range=[0, 1],
+            ),
+            fo.Mean("predictions.detections[]", expr=bbox_area),
+            fo.Std("predictions.detections[]", expr=bbox_area),
+            fo.Sum("predictions.detections", expr=F().length()),
+            fo.Values("id"),
+        ]
+
+        agg_dicts = [a._serialize() for a in aggregations]
+        also_aggregations = [fo.Aggregation._from_dict(d) for d in agg_dicts]
+
+        results = dataset.aggregate(aggregations)
+        also_results = dataset.aggregate(also_aggregations)
+        self.assertEqual(len(results), len(also_results))
+
```

Comment:
This is a good example of where lower level unit tests should be written. Aggregations are already decoupled class. There's really no need to set up a dataset and samples and call aggregate just to test serialization. The crux of what needs to be tested is whether or not the aggregations and also_aggregations are the same:

``` py
bbox_area = F("bounding_box")[2] * F("bounding_box")[3]
aggregations = [
    fo.Bounds("predictions.detections.confidence"),
    fo.Count(),
    fo.Count("predictions.detections"),
    fo.CountValues("predictions.detections.label"),
    fo.Distinct("predictions.detections.label"),
    fo.HistogramValues(
        "predictions.detections.confidence",
        bins=50,
        range=[0, 1],
    ),
    fo.Mean("predictions.detections[]", expr=bbox_area),
    fo.Std("predictions.detections[]", expr=bbox_area),
    fo.Sum("predictions.detections", expr=F().length()),
    fo.Values("id"),
]

for agg in aggregations:
    also_agg = fo.Aggregation._from_dict(agg._serialize())
    # In my branch, I added the following dunder method to 
    # the `Aggregation` class:
    # def __eq__(self, other):
    #     return isinstance(other , self._class__) and self._kwargs() == other._kwargs()
    assert agg == also_agg
```